pilot/pkg/model/extensions.go

```diff
@@ -173,7 +173,15 @@ type WasmPluginListenerInfo struct {
 	Class istionetworking.ListenerClass
 
 	// Service that WasmPlugins can attach to via targetRefs (optional)
-	Service *Service
+	Services []*Service
+}
+
+func (listenerInfo WasmPluginListenerInfo) AddService(service *Service) WasmPluginListenerInfo {
```

Comment:
nit: rename `AddService` to something like `WithService`. `Add` makes it sound like it mutates the existing object which it doesn't necessarily?

I guess sometimes it does which maybe makes it even more confusing